File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.py
class Solution:
    def nearestValidPoint(self, x, y, points):
        """
        type x: int
        type y: int
        type points: List[List[int]]
        rtype: int
        """
        dis = 10000
        res = -1
        for i in range(len(points)):
            if points[i][0] == x or points[i][1] == y:
                distance = abs(x - points[i][0]) + abs(y - points[i][1])
                if distance < dis:
                    dis = distance
                    res = i
        return res

File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.cpp
#include <vector>

using std::vector;

class Solution {
public:
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int dis = INT_MAX;
        int res = -1;
        for (int i = 0; i < points.size(); i++)
        {
            if (points[i][0] == x || points[i][1] == y)
            {
                int distance = abs(x - points[i][0]) + abs(y - points[i][1]);
                if (distance < dis)
                {
                    dis = distance;
                    res = i;
                }
            }
        }
        return res;
    }
};